src/IndexBuilder.php

```diff
@@ -22,16 +22,16 @@
 
 class IndexBuilder
 {
-    private $client;
-    private $configurationDirectory;
+    private Client $client;
+    private string $configurationDirectory;
 
-    public function __construct(Client $client, $configurationDirectory)
+    public function __construct(Client $client, string $configurationDirectory)
     {
         $this->client = $client;
         $this->configurationDirectory = $configurationDirectory;
     }
 
-    public function createIndex($indexName, $fileName = null): Index
+    public function createIndex(string $indexName, ?string $fileName = null): Index
```

Comment:
Note that adding parameter types is technically a breaking change, because it forces users extending this class to add them, too.